fairlearn/postprocessing/_threshold_optimizer.py

```diff
@@ -137,19 +157,42 @@ class ThresholdOptimizer(BaseEstimator, MetaEstimatorMixin):
                 allowed for all constraint types except 'equalized_odds'
 
     grid_size : int, default=1000
-        The values of the constraint metric are discretized according to the grid
-        of the specified size over the interval [0,1] and the optimization is
-        performed with respect to the constraints achieving those values. In case
-        of 'equalized_odds' the constraint metric is the false positive rate.
+        The values of the constraint metric are discretized according to the
+        grid of the specified size over the interval [0,1] and the optimization
+        is performed with respect to the constraints achieving those values. In
+        case of 'equalized_odds' the constraint metric is the false positive
+        rate.
 
     flip : bool, default=False
         If True, then allow flipping the decision if it improves the resulting
 
     prefit : bool, default=False
-        If True, avoid refitting the given estimator. Note that when used
-        with :func:`sklearn.model_selection.cross_val_score`,
-        :class:`sklearn.model_selection.GridSearchCV`, this will result in an error.
-        In that case, please use ``prefit=False``.
+        If True, avoid refitting the given estimator. Note that when used with
+        :func:`sklearn.model_selection.cross_val_score`,
+        :class:`sklearn.model_selection.GridSearchCV`, this will result in an
+        error. In that case, please use ``prefit=False``.
+        
+    predict_method : {'auto', 'predict_proba', 'decision_function', 'predict'\
+            }, default='auto'
+        Defines which method of the ``estimator`` is used to get the output
+        values.
+        - 'auto': use one of ``predict_proba``, ``decision_function``, or
+          ``predict``, in that order.
+        - 'predict_proba': use the second column from the putput of
+          `predict_proba`. It is assumed that the second column represents the
+          positive outcome.
+        - 'decision_function': use the raw values given by the
+          `decision_function`.
+        - 'predict': use the hard values reported by the `predict` method.
+          This is equivalent to what is done in [1]_.
+
+        .. versionadded:: 0.7
+            In previous versions only the ``predict`` method was used
+            implicitly.
+
+        .. versionchanged:: 0.7
+            From version 0.7, 'predict' is deprecated and the default will
+            change to 'auto' from v0.10.
```

Comment:
v0.10 is just arbitrarily chosen to be far enough out in the future or do we always want to do +0.3? 😆 
I'm just asking so that we have a consistent approach.

Reply:
I'm doing +3 versions here since we release pretty quickly. But yeah, we should be consistent kinda.